### PR TITLE
Nikhil2112

### PR DESCRIPTION
**Pull request by nikhilsamrat**
Created a Map from clientKey to id

### CODE REVIEW

File: beacon/include/marlin/beacon/Grapher.hpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		marlin::asyncio::TcpOutFiber,
+		marlin::core::DynamicFramingFiberHelper<
+			marlin::core::FabricF<SentinelFramingFiberHelper, marlin::core::SentinelBufferFiber>::type,
+			marlin::core::FabricF<marlin::core::LengthFramingFiber, marlin::core::LengthBufferFiber>::type
+		>::type
+	>;
+
+	marlin::asyncio::Timer t;
+	marlin::core::SocketAddress dst;
+	std::map< std::string, std::string> MData;
```
Comment (roshanr95) [inline beacon/include/marlin/beacon/Grapher.hpp:36]: give a better name and in snake_case e.g. client_key_map

File: beacon/include/marlin/beacon/Grapher.hpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+	void query_cb() {
```
Comment (roshanr95) [inline beacon/include/marlin/beacon/Grapher.hpp:38]: name it query_subgraph

File: beacon/include/marlin/beacon/Grapher.hpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+				std::make_tuple()
+			))
+		));
+		(void)fiber.i(*this).dial(dst);
+	}
+
+	template<typename... Args>
+	Grapher(Args&&...) : t(this) {
+		//query();
+		t.template start<Grapher, &Grapher::query>(0, 3000000);
```
Comment (roshanr95) [inline beacon/include/marlin/beacon/Grapher.hpp:54]: extra 0, should be 300k

File: beacon/include/marlin/beacon/ClusterDiscoverer.hpp [hunk 4]
```diff
@@ -5,13 +5,23 @@
...
 #include <marlin/core/fabric/Fabric.hpp>
+#include <marlin/beacon/Grapher.hpp>
 #include <map>
 
 #include <sodium.h>
 
 #include "Messages.hpp"
 #include "PeerInfo.hpp"
 
+uint8_t* hextoint(std::string s){
```
Comment (roshanr95) [inline beacon/include/marlin/beacon/ClusterDiscoverer.hpp]: add `static`

File: beacon/include/marlin/beacon/ClusterDiscoverer.hpp [hunk 5]
```diff
@@ -5,13 +5,23 @@
...
+	//hex strings starting with 0x//
+	uint8_t* res = new uint8_t[(s.length()-2)*2];
```
Comment (roshanr95) [inline beacon/include/marlin/beacon/ClusterDiscoverer.hpp:18]: memory leak here

File: beacon/include/marlin/beacon/Grapher.hpp [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	int did_close(auto& fiber) {
+		SPDLOG_DEBUG("Did close");
+		delete &fiber;
+		return 0;
+	}
+
+    void query() {
+        uv_getaddrinfo_t* req = new uv_getaddrinfo_t();
```
Comment (roshanr95) [inline beacon/include/marlin/beacon/Grapher.hpp]: memory leak here

**Review by roshanr95 — CHANGES_REQUESTED** [review]

File: beacon/include/marlin/beacon/Grapher.hpp (hunk 6 above)
Comment (nikhilsamrat) [reply beacon/include/marlin/beacon/Grapher.hpp]: That is required.

File: beacon/include/marlin/beacon/ClusterDiscoverer.hpp (hunk 5 above)
Comment (nikhilsamrat) [reply beacon/include/marlin/beacon/ClusterDiscoverer.hpp:18]: resolved in next commit

File: beacon/include/marlin/beacon/ClusterDiscoverer.hpp (hunk 4 above)
Comment (nikhilsamrat) [reply beacon/include/marlin/beacon/ClusterDiscoverer.hpp]: resolved

File: beacon/include/marlin/beacon/Grapher.hpp (hunk 3 above)
Comment (nikhilsamrat) [reply beacon/include/marlin/beacon/Grapher.hpp:54]: resolved in next commit

File: beacon/include/marlin/beacon/Grapher.hpp (hunk 2 above)
Comment (nikhilsamrat) [reply beacon/include/marlin/beacon/Grapher.hpp:38]: resolved in next commit

File: beacon/include/marlin/beacon/Grapher.hpp (hunk 6 above)
Comment (nikhilsamrat) [reply beacon/include/marlin/beacon/Grapher.hpp]: deleted in the callback

**Review by roshanr95 — APPROVED** [review]

**Pull request merged**